src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -291,11 +260,32 @@ private void BoundsCheck(SequencePosition start, SequencePosition position)
             }
         }
 
-        private class ReadOnlySequenceSegment : IMemoryList<T>
+        private sealed class ReadOnlySequenceSegment : IMemoryList<T>
         {
             public Memory<T> Memory { get; set; }
-            public IMemoryList<T> Next { get; set; }
-            public long RunningIndex { get; set; }
+            public IMemoryList<T> Next { get; } = null;
+
+            public IMemoryList<T> GetNext(long offset, out int localOffset)
+            {
+                if (offset <= Memory.Length)
+                {
+                    localOffset = (int)offset;
+                    return this;
+                }
+
+                localOffset = 0;
+                return null;
+            }
+
+            public long GetLength(IMemoryList<T> memoryList)
+            {
+                if (memoryList != this)
+                {
+                    ThrowHelper.ThrowInvalidOperationException();
+                }
+
+                return 0;
```

Comment:
Should this return Memory.Length?

Reply:
No, this method calculates distance between two IMemoryList segments, distance to self if 0.

Reply:
Can you add that comment in code to clarify. Thanks.